odoo/fields.py

```diff
@@ -1752,6 +1754,14 @@ def convert_to_column(self, value, record, values=None, validate=True):
         # on purpose - non base64 data must be passed as a 8bit byte strings.
         if not value:
             return None
+        # Detect if the binary content is an SVG for restricting its upload
+        # only to system users.
+        if value[:1] == b'P':  # Fast detection of first 6 bits of '<' (0x3C)
+            decoded_value = base64.b64decode(value)
+            # Full mimetype detection
+            if (guess_mimetype(decoded_value) == 'image/svg+xml' and
```

Comment:
We should use `guess_mimetype(decoded_value).startswith('image/svg')` to be on the safe side, as the python-magic library returns `image/svg` instead of `image/svg+xml`. There's a remote chance that some browsers will render `image/svg` as a real SVG.

Reply:
... and it could be used [instead of the built-in detection](https://github.com/odoo/odoo/blob/8c9a621227aad6f02cccf4cd4f3d61f02531cddd/odoo/tools/mimetypes.py#L164-L178) if `python-magic` is installed.

Reply:
Actually it depends on whether there is a XML header in the SVG file. When present, the detected mimetype is image/svg+xml. This is the same behavior as the `file` unix command, as it uses the same underlying library.
I haven't tested yet how browsers behave with the short version, hopefully they don't render it as full SVG. In that case it's safe ; worst case would be that you can't use SVG fully if you have installed python-magic (but no security risk apparently)

Reply:
OK, no problem is restricting this a bit more.